network/p2p/throttler_handler.go

```diff
@@ -20,25 +20,37 @@ var (
 	_            Handler = (*ThrottlerHandler)(nil)
 )
 
+func NewThrottlerHandler(handler Handler, throttler Throttler, log logging.Logger) *ThrottlerHandler {
+	return &ThrottlerHandler{
+		handler:   handler,
+		throttler: throttler,
+		log:       log,
+	}
+}
+
 type ThrottlerHandler struct {
-	Handler
-	Throttler Throttler
-	Log       logging.Logger
+	handler   Handler
+	throttler Throttler
+	log       logging.Logger
 }
 
 func (t ThrottlerHandler) AppGossip(ctx context.Context, nodeID ids.NodeID, gossipBytes []byte) {
-	if !t.Throttler.Handle(nodeID) {
-		t.Log.Debug("dropping message", zap.Stringer("nodeID", nodeID))
+	if !t.throttler.Handle(nodeID) {
+		t.log.Debug("dropping message", zap.Stringer("nodeID", nodeID))
```

Comment:
also could be nice to have drop reason here

Reply:
might be good to use same syntax with `AppRequest`

Reply:
I added a log to both. We don't log this in `AppRequest` because it's bubbled up into `responder` where the error is logged.